doc/hardware/porting/board_porting.rst

```diff
@@ -354,7 +354,8 @@ named ``plank``:
 
 :file:`plank_defconfig`
   A Kconfig fragment that is merged as-is into the final build directory
-  :file:`.config` whenever an application is compiled for your board.
+  :file:`.config` whenever an application is compiled for your board. The name
+  of this file also determines the name of the board as seen by the build system.
```

Comment:
Not precise.

The build system will use the `BOARD` provided by the user with `-DBOARD=<board>` (or `-b` with `west build`).

From there the build system looks up board related files, such as `<board>_defconfig` for Kconfig, `<board>.dts` for devicetree, `doc/<board>.png` for documentation, `<board>.yaml` for twister.

In an upstream board all those files are required, but for a downstream board only the Kconfig is mandatory, cause a downstream board doesn't have to use devicetree (although most does).

This is why the check for a `_defconfig` is the first thing that is checked, because if this file is missing, the user has typed a wrong board or the board impl is wrong.
Ref: https://github.com/zephyrproject-rtos/zephyr/blob/147d9cb1272b39e119025953acdb3d19ba52e45e/cmake/modules/boards.cmake#L113-L114

Also because the `_defconfig` is the only mandatory file, then `west boards` uses this file to search all valid board names.

I agree, the fact that the name of the board dir doesn't need to be identical the board itself can cause confusion, for example like: `nrf5340dk_nrf5340` dir contains the boards `nrf5340dk_nrf5340_cpunet`, `nrf5340dk_nrf5340_cpuapp`, `nrf5340dk_nrf5340_cpuapp_ns`.
But this should be fixed by improving the overall description of the board folder vs. the actual board, not by making imprecise description of the files describing the actual board.